contributing-docs/17_adding_api_endpoints.rst

```diff
@@ -18,122 +18,78 @@
 Adding a New API Endpoint in Apache Airflow
 ===========================================
 
-This documentation outlines the steps required to add a new API endpoint in Apache Airflow. It includes defining the endpoint in the OpenAPI specification, implementing the logic, running pre-commit checks, and documenting the changes.
+This documentation outlines the steps required to add a new API endpoint in Apache Airflow. It includes implementing the logic, running pre-commit checks, and documenting the changes.
```

Comment:
Let's include something about which directory to now contribute API code to?

Reply:
I like the idea of having this in the doc. 

Should we include something about the compatibility difference between `public/` and `ui/` directories somewhere in the contributing docs? Even though it's [documented in Confluence at a high](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=315493907#AIP84UIRESTAPI-Proposal), it could be beneficial for everyone to know what compatibility rules are defined while making changes in API endpoints. 
Something like,
`ui/ -> don't have to be backwards compatible`
`public/ -> have to be backwards compatible`

Reply:
> Let's include something about which directory to now contribute API code to?

Good idea, done

Reply:
@bugraoz93 Good idea, done!